File: MediaSpaceBackend/space-service/src/main/java/tn/medianet/mediaspace/spaceservice/entity/Space.java
package tn.medianet.mediaspace.spaceservice.entity;


import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Space {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @OneToOne(cascade = CascadeType.ALL)
    private Address address;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false)
    private double price;
    private int maxGuest;
    private int roomNumber;
    private int bathroomNumber;
    private int floorNumber;
    private int restrictedMinAge;
    private int restrictedMaxAge;
    private SpaceType spaceType;
    @Column(length = 5000)
    private String spaceRule;
    @ElementCollection
    private List<String> amenities;
    @ManyToMany(cascade = CascadeType.ALL)
    private List<SpaceEquipement> equipments;
    @ElementCollection
    private List<String> accessibility;
    @ElementCollection
    private List<String> images;
    @Column(nullable = false)
    private double squareFootage;
    @ElementCollection
    private List<SpaceAvailability> availabilities;

    @Column(name = "user_id", nullable = false)
    private Long ownerId;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;


}

File: MediaSpaceFrontend/lib/services/models/space.dart


import 'package:MediaSpaceFrontend/services/models/space_event_price.dart';

import 'Address.dart';
import 'SpaceEquipement.dart';
import 'SpaceType.dart';
import 'spaceavailability.dart';

class Space {
  int? id;
  String? name;
  Address? address;
  String? description;
  int? maxGuest;
  int? roomNumber;
  int? bathroomNumber;
  int? floorNumber;
  int? restrictedMinAge;
  int? restrictedMaxAge;
  SpaceType? spaceType;
  String? spaceRule;
  List<String>? amenities;
  List<SpaceEquipement>? equipments;
  List<String>? accessibility;
  List<String>? images;
  double? squareFootage;
  List<SpaceAvailability>? availabilities;
  int? ownerId;
  List<SpaceEventPrice>? eventPrices;
  DateTime? createdAt;
  DateTime? updatedAt;

  Space({
    this.id,
    this.name,
    this.address,
    this.description,
    this.maxGuest,
    this.roomNumber,
    this.bathroomNumber,
    this.floorNumber,
    this.restrictedMinAge,
    this.restrictedMaxAge,
    this.spaceType,
    this.spaceRule,
    this.amenities,
    this.equipments,
    this.accessibility,
    this.images,
    this.squareFootage,
    this.availabilities,
    this.ownerId,
    this.eventPrices,
    this.createdAt,
    this.updatedAt,
  });
}
